File: DCAnalyticsMobile/DCAnalyticsMobile/Models/UserRight.cs
using DCAnalytics;
using System;
using System.Collections.Generic;
using System.Text;

namespace DCAnalyticsMobile.Models
{
    public class UserRight : DataCollectionObject
    {
        public string ObjectName { get; set; }
        public ObjectType ObjectType { get; set; }
        public string PrimaryKey { get; set; }
        public Configuration Configuration { get; set; }
        public List<UserPermission> UserPermissions { get; set; }
    }
}

File: DCAnalyticsWebShell/ClientApp/src/models/UserRight.ts
import { ObjectType } from "src/enums/enums";
import { UserPermission } from "./UserPermission";
import { Configuration } from "./Configuration";

export class UserRight {
  ObjectName: string
  ObjectType: ObjectType
  PrimaryKey: string
  Deleted: boolean
  SecondaryKey: string
  IsSystem: boolean
  Key: string
  UserPermissions: UserPermission[]
  Configuration: Configuration
}
